bebraven/platform
# Lessons send xAPI statements to LRS

**Task**: https://app.asana.com/0/1174274412967132/1183556772940371

**Dev environment breaking changes**: 

- Rebuild: `rack-cors` gem added, you'll need to rebuild you Docker container.

**Test**:  

```
rspec spec/helpers/lesson_contents_helper.rb
rspec spec/controllers/lesson_contents_controller.spec.rb
```

**If you're running into login issues, you can serve a static lesson from local `platformweb` instead of AWS S3 using this [commit](https://github.com/bebraven/platform/commit/96540a7c833ed4e38a5ad3e623ad6a29ceab33d3), or by running**:
```
git cherry-pick -n 96540a7c833ed4e38a5ad3e623ad6a29ceab33d3
```
FYI, it'll cause the controller spec to fail, since we're altering the redirect. 

 - Login to `platformweb`
 - Upload a Rise 360 course (see PR #279 for instructions)
 - Open the link returned (something like, https://platformweb/lesson_contents/1) in a new browser tab
 - See that there are xAPI events logged in Learning Locker (https://lrs.braven.org/):
<img width="1792" alt="Screen Shot 2020-07-23 at 2 58 30 PM" src="https://user-images.githubusercontent.com/62911934/88342876-1afc0800-ccf5-11ea-9a42-d909da365bfe.png">
 - You can also look at the network traffic in the Chrome developer console and verify that `statements/` work:
<img width="1792" alt="Screen Shot 2020-07-23 at 2 58 48 PM" src="https://user-images.githubusercontent.com/62911934/88342880-1d5e6200-ccf5-11ea-8ce3-de672bb468a0.png">
